sklearn/metrics/cluster/unsupervised.py

```diff
@@ -201,3 +206,160 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
+        List of n_features-dimensional data points. Each row corresponds
+        to a single data point.
+
+    labels : array-like, shape (n_samples,)
+        Predicted labels for each sample.
+
+    Return
+    ------
+    score: float
+        The resulting Calinski-Harabaz score.
+
+    References
+    ----------
+    .. [1] `T. Calinski and J. Harabasz, 1974. "A dendrite method for cluster
+       analysis". Communications in Statistics
+       <http://www.tandfonline.com/doi/abs/10.1080/03610927408827101>`_
+    """
+    X, labels = check_X_y(X, labels)
+    le = LabelEncoder()
+    labels = le.fit_transform(labels)
+
+    n_samples, _ = X.shape
+    n_labels = len(le.classes_)
+
+    check_number_of_labels(n_labels, n_samples)
+
+    extra_disp, intra_disp = 0., 0.
+    mean = np.mean(X, axis=0)
+    for k in range(n_labels):
+        cluster_k = X[labels == k]
+        mean_k = np.mean(cluster_k, axis=0)
+        extra_disp += len(cluster_k) * np.sum((mean_k - mean) ** 2)
+        intra_disp += np.sum((cluster_k - mean_k) ** 2)
+
+    return (1. if intra_disp == 0. else
+            extra_disp * (n_samples - n_labels) /
+            (intra_disp * (n_labels - 1.)))
+
+
+def distortion_score(X, labels, metric="euclidean", **kwds):
+    """Compute the distortion of a given dataset and their cluster assignment.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        List of n_features-dimensional data points. Each row corresponds
+        to a single data point.
+
+    labels : array-like, shape (n_samples,)
+        Predicted labels for each sample.
+
+    metric : string, or callable
+        The metric to use when calculating distance between instances in a
+        feature array. If metric is a string, it must be one of the options
+        allowed by :func:`sklearn.metrics.pairwise.pairwise_distances`.
+
+    **kwds : optional keyword parameters
+        Any further parameters are passed directly to the distance function.
+        If using a scipy.spatial.distance metric, the parameters are still
+        metric dependent. See the scipy docs for usage examples.
+
+    Returns
+    -------
+    score: float
+        The resulting distortion value.
+    """
+    X, labels = check_X_y(X, labels)
+    le = LabelEncoder()
+    labels = le.fit_transform(labels)
+
+    n_samples, n_features = X.shape
+    n_labels = len(le.classes_)
+
+    check_number_of_labels(n_labels, n_samples)
+
+    dist = 0.
+    for k in range(n_labels):
+        cluster_k = X[labels == k]
+        mean_k = np.mean(cluster_k, axis=0)
+        dist += np.sum(pairwise_distances(cluster_k,
+                                          mean_k[:, np.newaxis].reshape(1, -1),
+                                          metric=metric, **kwds))
+
+    return dist / n_features
+
+
+def inertia_score(X, labels, metric="euclidean", **kwds):
+    """Compute the inertia of a given dataset and their cluster assignment.
```

Comment:
I've added also `inertia` because I will need it later to implement the gap statistics.
To be honest, I don't see what are the differences between distortion and inertia (unless the normalization).

I know Inertia is the function which is minimized by kmeans but is it really useful to have in a specific function like this ? Tell me if you want to keep it or not.